src/settings/LocationLocations/LocationManager.js

```diff
@@ -459,58 +460,50 @@ class LocationManager extends React.Component {
     this.transitionToParams({ layer: 'clone' });
   };
 
+  checkLocationHasHoldingsOrItems = async (locationId) => {
+    const { mutator } = this.props;
+    const query = `permanentLocationId=="${locationId}" or temporaryLocationId=="${locationId}"`;
+
+    mutator.holdingsEntries.reset();
+    mutator.itemEntries.reset();
+
+    const results = await Promise.all([
+      mutator.holdingsEntries.GET({ params: { query } }),
+      mutator.itemEntries.GET({ params: { query } }),
+    ]);
+
+    return results.some(records => records.length > 0);
+  };
+
   onRemove = location => {
     const {
       match,
       mutator,
     } = this.props;
-    const promises = [];
-
-    // Uses in Inventory
-    {
-      mutator.holdingsEntries.reset();
-      mutator.itemEntries.reset();
-      const query = `permanentLocationId=${location.id} or temporaryLocationId=${location.id}`;
-      promises.push(mutator.holdingsEntries.GET({ params: { query } }));
-      promises.push(mutator.itemEntries.GET({ params: { query } }));
-    }
 
-    return Promise.all(promises)
-      .then(values => {
-        if (undefined === values.find(records => records.length !== 0)) {
-          return mutator.entries.DELETE(location);
-        }
+    return this.checkLocationHasHoldingsOrItems(location.id)
+      .then(itHas => !itHas && mutator.entries.DELETE(location))
```

Comment:
```suggestion
      .then(result => !result && mutator.entries.DELETE(location))
```

Reply:
Isn't `itHas` more self-descriptive?

Reply:
Fair enough. Usually, boolean variables are named starting with the verb e.g. isValid, hasResult. I would suggest then renaming this variable to something which starts with `has`. Frankly speaking, I could not come up with such a name. Leaving this up to you.

Reply:
done